server/src/main/java/org/candlepin/resource/util/ConsumerBindUtil.java

```diff
@@ -95,12 +97,18 @@ public void handleActivationKeys(Consumer consumer, List<ActivationKey> keys,
             handleActivationKeyAddons(consumer, key.getAddOns());
 
             if (key.isAutoAttach() != null && key.isAutoAttach()) {
-                if (autoattachDisabledForOwner || key.getOwner().isContentAccessEnabled()) {
+                if (autoattachDisabledForOwner || key.getOwner().isContentAccessEnabled() || !isAutoheal) {
                     String caMessage = "";
                     if (key.getOwner().isContentAccessEnabled()) {
                         caMessage = " because of the content access mode setting";
                         isCAModeEnabledForAny = true;
                     }
+                    if (!isAutoheal) {
+                        caMessage = StringUtils.isBlank(caMessage) ?
+                            " because of the consumer autoheal value" :
+                            " and consumer autoheal value";
```

Comment:
I know this is a bit nit-picky, but good log messages help people figure out what's happened without having to attach a debugger. Using the same warning log output message and appending either of these two would result in misleading warning messages:

On the scenario where contentAccessEnabled is true, this is a bit misleading:
`Auto-attach is disabled for owner because of the content access mode setting and consumer autoheal value. Skipping...`
It should be more like:
`Auto-attach is disabled for owner because of the content access mode setting. Also, the consumer autoheal value is false. Skipping...`

On the scenario where contentAccessEnabled is false, this is very misleading:
`Auto-attach is disabled for owner because of the consumer autoheal value. Skipping...`
The consumer autoheal value has no effect on the owner's auto-attach setting, so auto-attach cannot be disabled for the owner. This should be more like:
`The consumer autoheal value is false. Skipping...`

Reply:
@nikosmoum thanks for catching this, earlier it was absolutely incorrect. I have fixed it. Please re-review.